book/src/dev/rfcs/0011-value-pool.md

```diff
@@ -0,0 +1,229 @@
+
+# Definitions
+[definitions]: #definitions
+
+- `transparent pool` - Sum of all `tx_in` fields for transactions in the block chain, minus the sum of all `tx_out` fields for transactions in the block chain.
+- `sprout pool` - Sum of all `vpub_old` fields for transactions in the block chain, minus the sum of all `vpub_new` fields for transactions in the block chain [ZIP-209](https://zips.z.cash/zip-0209#terminology).
+- `sapling pool` - Negation of the sum of all `valueBalanceSapling` fields for transactions in the block chain [ZIP-209](https://zips.z.cash/zip-0209#terminology).
+- `orchard pool` - Negation of the sum of all `valueBalanceOrchard` fields for transactions in the block chain.
+- `transaction pool`
+- `block pool`
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Each of the 4 value pools change its value with every upcoming block. This is a state feature and Zebra handle this in the `zebra-state` crate. We propose to store the pool values among a reference block height into the disk.
+
+The proposed design goes from calculating transaction pools, summing them all to form block pools and finally summing all block pools to form the final pool value.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Changes to `zebra-chain/src/transaction.rs`
+
+We have a function to calculate each transaction pool for any transaction version:
+
+### `transparent_value_pool()`
+
+TODO
+
+### `sprout_value_pool()`
+
+```rust
+pub fn sprout_value_pool(&self) -> Amount {
+    match self {
+        Transaction::V1 { .. } => Amount::try_from(0).unwrap(),
+        Transaction::V2 { joinsplit_data, .. } |
+        Transaction::V3 { joinsplit_data, .. } => {
+            let mut value_balance: Amount<NegativeAllowed> = Amount::try_from(0).unwrap();
+            for js in joinsplit_data.as_ref().unwrap().joinsplits() {
+                value_balance = (value_balance + (js.vpub_old - js.vpub_new).unwrap().constrain()).unwrap();
+            }
+            value_balance
+        },
+        Transaction::V4 { joinsplit_data, .. } => {
+            let mut value_balance: Amount<NegativeAllowed> = Amount::try_from(0).unwrap();
+            for js in joinsplit_data.as_ref().unwrap().joinsplits() {
+                value_balance = (value_balance + (js.vpub_old - js.vpub_new).unwrap().constrain()).unwrap();
+            }
+            value_balance
+        },
+        Transaction::V5 { .. } => Amount::try_from(0).unwrap(),
+    }
+}
+```
+
+### `sapling_value_pool()`
+
+```rust
+pub fn sapling_value_pool(&self) -> Amount {
+    match self {
+        Transaction::V1 { .. } => Amount::try_from(0).unwrap(),
+        Transaction::V2 { .. } => Amount::try_from(0).unwrap(),
+        Transaction::V3 { .. } => Amount::try_from(0).unwrap(),
+        Transaction::V4 { value_balance, .. } => *value_balance,
```

Comment:
This can change if https://github.com/ZcashFoundation/zebra/pull/1886 is merged as it is because proposes to move `value_balance` into `ShieldedData::value_balance`